src/service/plugins/contacts.js

```diff
@@ -146,12 +146,15 @@ var Plugin = GObject.registerClass({
     /**
      * Decode a string encoded as "QUOTED-PRINTABLE" and return a regular string
      *
-     * See: https://github.com/mathiasbynens/quoted-printable/blob/master/src/quoted-printable.js
-     *
      * @param {string} input - The QUOTED-PRINTABLE string
      * @return {string} The decoded string
      */
     _decodeQuotedPrintable(input) {
+        /* SPDX-SnippetBegin
+         * SPDX-License-Identifier: MIT
+         * SPDX-SnippetCopyrightText: Mathias Bynens
+         * SPDX-SnippetComment: https://github.com/mathiasbynens/quoted-printable/blob/master/src/quoted-printable.js
+         */
```

Comment:
For instance, this file should probably cause the `reuse lint` check to fail, because the MIT license is not currently installed in `/LICENSES/`. The path to that would seem to be making the header something like,

```js
// SPDX-FileCopyrightText: GSConnect Developers https://github.com/GSConnect
//
// SPDX-License-Identifier: GPL-2.0-or-later
// 
// License for _decodeQuotedPrintable():
// SPDX-License-Identifier: MIT
// SPDX-SnippetCopyrightText: Mathias Bynens
// SPDX-SnippetComment: https://github.com/mathiasbynens/quoted-printable/blob/master/src/quoted-printable.js
```

...I think that could work... (Well, I just tested, and it definitely **does** work to make `reuse lint` fail until the MIT license file is downloaded into the repo.)

Whether we then still duplicate all of that info down here, or limit the local comment to the `SPDX-Snippet...` markers and a reference to the header. Either approach seems pretty reasonable to me.

Reply:
Either's fine by me. REUSE won't find things it doesn't know about anyways, so having the snippet outside the header seems relatively moot.

There's also `SDPX-SnippetLineRange: <start>:<finish>` that can be used in the header, although REUSE probably doesn't recognize that field either.